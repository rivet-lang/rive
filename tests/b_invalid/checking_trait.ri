trait ToStr {
    proc to_str1(&self) -> string;
}

proc my_print(v: ToStr) {
    _ = v.to_str1();
}

extend string < ToStr {
    proc to_str1(self) -> string {
        return self;
    }
}

extend bool < ToStr {
    proc to_str1(&self) -> string {
        return if self.* { "true" } else { "false" };
    }
}

proc main() {
    my_print(""); // OK
    my_print(true); // OK
    my_print(1); // FAIL
}
