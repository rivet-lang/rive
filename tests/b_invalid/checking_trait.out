tests/b_invalid/checking_trait.ri:24:14: error: type `int` does not implement trait `ToStr`
   24 |     my_print(1); // FAIL
      |              ^
      = note: in argument `v` of procedure `my_print`
rivetc: error: could not compile module `checking_trait`, aborting due to previous error